### PR TITLE
Add `rust-run-clippy' and `rust-buffer-project' with testing paraphernalia

### PR DESCRIPTION
**Pull request by Aankhen**
Adds a `rust-run-clippy` command to run `cargo clippy` using `compile`. To support this, I added a `rust-buffer-project` function that uses `cargo locate-project` to find the manifest for the buffer.

### CODE REVIEW

File: test-project/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+1
```
Comment (nikomatsakis) [inline test-project/Cargo.toml]: what is this?

Comment (Aankhen) [inline test-project/Cargo.toml]: `rust-buffer-project` uses `cargo locate-project`. The test for it doesn’t need anything beyond the presence of Cargo.toml in the directory, so I didn’t put anything in it. If that’s misleading I can put in some sample content.

Comment (nikomatsakis) [inline test-project/Cargo.toml]: Maybe a comment like "# Dummy file needed for test"

**Comment by Aankhen** [comment]
Thanks for the review, @nikomatsakis. I’ve made the change.

**Comment by nikomatsakis** [comment]
OK, well, I'm inclined to merge =) cc @tromey -- care to weigh in?

**Comment by tromey** [comment]
Sorry about the delay on this.  It seems fine to me.

Note that flycheck has a pending patch to get clippy support: https://github.com/flycheck/flycheck/pull/1156

**Pull request merged**